Content/default/src/Client/Index.fs

```diff
@@ -5,89 +5,94 @@ open SAFE
 open Shared
 
 type Model = {
-    Todos: Deferred<Todo list>
+    Todos: RemoteData<Todo list>
     Input: string
 }
 
 type Msg =
     | SetInput of string
-    | GetTodos of AsyncOperation<unit, Todo list>
-    | AddTodo of AsyncOperation<unit, Todo>
+    | LoadTodos of ApiCall<unit, Todo list>
+    | SaveTodo of ApiCall<string, Todo>
 
 let todosApi = Api.makeProxy<ITodosApi> ()
 
 let init () =
-    let model = { Todos = NotStarted; Input = "" }
-    model, Cmd.ofMsg (GetTodos(Start()))
+    let initialModel = { Todos = NotStarted; Input = "" }
+    let initialCmd = Cmd.ofMsg (LoadTodos(Start()))
+
+    initialModel, initialCmd
 
 let update msg model =
     match msg with
     | SetInput value -> { model with Input = value }, Cmd.none
-    | GetTodos msg ->
+    | LoadTodos msg ->
         match msg with
         | Start() ->
-            let cmd = Cmd.OfAsync.perform todosApi.getTodos () (Finished >> GetTodos)
-            { model with Todos = InProgress }, cmd
-        | Finished todos -> { model with Todos = Resolved todos }, Cmd.none
-    | AddTodo msg ->
+            let loadTodosCmd = todosApi.getTodos () |> Async.toCmdUnsafe LoadTodos
+            { model with Todos = Loading }, loadTodosCmd
+        | Finished todos -> { model with Todos = Loaded todos }, Cmd.none
+    | SaveTodo msg ->
         match msg with
-        | Start() ->
-            let todo = Todo.create model.Input
-            let cmd = Cmd.OfAsync.perform todosApi.addTodo todo (Finished >> AddTodo)
-            { model with Input = "" }, cmd
+        | Start todoText ->
+            let saveTodoCmd =
+                let todo = Todo.create todoText
+                todosApi.addTodo todo |> Async.toCmdUnsafe SaveTodo
```

Comment:
I think that `Async.toApiCallCmd` might be a better name than `Async.toCmdUnsafe`. The current function implies to me more generic async behaviour than the specific `ApiCall` behaviour it actually represents.

In what sense is the function unsafe?

Reply:
It doesn't handle exceptions. There's a `toCmd` variant which requires two messages e.g.

```fsharp
todosApi.addTodo todo |> Async.toCmd SaveTodo ApiError
```

or similar.

There's XML docs around this, but perhaps a different name?

I would like to encourage people to always include an error handler - too many errors get silently swallowed otherwise.

Reply:
Regarding the ApiCall behaviour - it will actually work with any Async workflow, although you're right, it only works with the `ApiCall` type as it automatically wraps it in `Finished`. I also considered:

`ApiCall.dispatch`
`ApiCall.toCmd` (but this is misleading to me, we're not going from an ApiCall to a Cmd)

Your suggestion of `Async.toApiCallCmd` is the most accurate, but it feels like a bit of a mouthful?

Reply:
How about `Cmd.OfAsyncApiCall.xxx` where `xxx` is some meaningful name e.g.

* `execute`
* `start`
* `executeSafe`

Open to ideas :)

Reply:
I'm currently thinking that we just shouldn't include this function at all :sweat_smile: (see other thread)

Happy to consider names if/when we agree to add this.